xdsl/dialects/test.py

```diff
@@ -0,0 +1,50 @@
+from __future__ import annotations
+from xdsl.ir import Attribute, Data, Dialect, OpResult, Operation, TypeAttribute
+
+from xdsl.irdl import VarOpResult, VarOperand, VarRegion, irdl_attr_definition, irdl_op_definition
+from xdsl.parser import BaseParser
+from xdsl.printer import Printer
+
+
+@irdl_op_definition
+class TestOp(Operation):
+    """
+    This operation can produce an arbitrary number of SSAValues with arbitrary
+    types. It is used in filecheck testing to reduce to artificial dependencies
+    on other dialects (i.e. dependencies that only come from the structure of
+    the test rather than the actual dialect).
+    """
+    name: str = "test.op"
+
+    res: VarOpResult
+    ops: VarOperand
+    regs: VarRegion
+
+    @staticmethod
+    def from_result_types(*res: Attribute) -> TestOp:
+        return TestOp.create(result_types=res)
```

Comment:
Do you even need those? I guess they could come in handy when porting pytests, but that could be added once we are there.

Reply:
Yeah I guess you are right, it's better to keep this to a minimum for now, we can add this later (especially since Sasha added TestSSAValue that might remove the need for the test dialect in pytest).